osu.Game.Rulesets.Taiko/Difficulty/Evaluators/ReadingEvaluator.cs

```diff
@@ -31,13 +32,28 @@ public VelocityRange(double min, double max)
         /// <returns>The reading difficulty value for the given hit object.</returns>
         public static double EvaluateDifficultyOf(TaikoDifficultyHitObject noteObject)
         {
-            double effectiveBPM = noteObject.EffectiveBPM;
+            double effectiveBPM = Math.Max(1.0, noteObject.EffectiveBPM);
+			// Expected deltatime is the deltatime this note would need
+			// to be spaced equally to a base SV 1/4 note
+			double expectedDeltaTime = 21000.0 / effectiveBPM;
 
+			var midVelocity = new VelocityRange(360, 480);
             var highVelocity = new VelocityRange(480, 640);
-            var midVelocity = new VelocityRange(360, 480);
+			
+			double midVelDifficulty = 0.5 * DifficultyCalculationUtils.Logistic(effectiveBPM, midVelocity.Center, 1.0 / (midVelocity.Range / 10));
 
-            return 1.0 * DifficultyCalculationUtils.Logistic(effectiveBPM, highVelocity.Center, 1.0 / (highVelocity.Range / 10))
-                   + 0.5 * DifficultyCalculationUtils.Logistic(effectiveBPM, midVelocity.Center, 1.0 / (midVelocity.Range / 10));
+			// Density refers to an object's deltatime relative to its expected deltatime
+			double density = expectedDeltaTime / Math.Max(1.0, noteObject.DeltaTime);
+			
+			// Dense notes are penalised at high velocities
```

Comment:
For above - call `density` `objectDensity` instead.

Maybe just clean up this comment a bit too, maybe 

`High densities are penalised at high effective bpm values as they are generally considered easier to read.`